src/D2L.Bmx/Aws/AwsRoleState.cs

```diff
@@ -0,0 +1,13 @@
+namespace D2L.Bmx.Aws;
+
+internal class AwsRoleState {
+	public string?[] Roles { get; }
+	internal List<AwsRole> AwsRoles { get; }
+	internal string SamlString { get; }
+
+	public AwsRoleState( List<AwsRole> awsRoles, string samlString ) {
+		AwsRoles = awsRoles;
+		SamlString = samlString;
+		Roles = AwsRoles.Select( role => role.RoleName ).ToArray();
+	}
```

Comment:
* All "public" members inside an internal class are effectively "internal", so people typically mark them as "public" instead of "internal"
* If `Roles` are derived from `AwsRoles`, do we really need `Roles`? Can we we simplify this type to just two properties and use record with primary constructor?

Reply:
oh! I see you've actually made that change in your "okta" PR.
would be better if that's in this PR, but this is fine too.

Reply:
So keep AwsRoles as a property and separate Roles out as it's own record?

Reply:
No, I meant deleting the `Roles` property entirely.
So what you did here is good I think:
https://github.com/Brightspace/bmx/pull/239/files#diff-5457c3da07d8bb435dcdf1616bc3b67556b5b83a0454143321babfebdaa61346